### PR TITLE
Enable tls connection with mongo server

### LINKED ISSUE #18: How to we configure it with ssl ?
I just wonder how we could configure it with ssl options ?

Also isn't this exporting compatible with rocksdb :cold_sweat: ?

### PR DESCRIPTION
**Pull request by ese**
Based on https://github.com/dcu/mongodb_exporter/pull/18
Fix #18

### CODE REVIEW

**Comment by timvaillancourt** [comment]
I was in the process of merging this same PR, it works great!

I will run this through some testing before approving.

**Comment by timvaillancourt** [comment]
Testing complete, no problems noticed. We will merge soon. Thanks

File: shared/connection.go [hunk 1]
```diff
@@ -48,6 +68,74 @@ func MongoSession(uri string) *mgo.Session {
...
+		}
+		dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
+			conn, err := tls.Dial("tcp", addr.String(), config)
+			if err != nil {
+				glog.Infof("Could not connect to %v. Got: %v", addr, err)
+				return nil, err
+			}
+			if config.InsecureSkipVerify {
+				err = enrichWithOwnChecks(conn, config)
+				if err != nil {
```
Comment (arvenil) [inline shared/connection.go]: Is this correct? Shouldn't this `return nil, err`? `enrichWithOwnChecks` returns closed connection on error. Returning error and closed connection might be misleading? This is not affecting anything so 👍

File: shared/connection.go [hunk 2]
```diff
@@ -37,9 +51,15 @@ func MongoSession(uri string) *mgo.Session {
 	dialInfo.Direct = true // Force direct connection
 	dialInfo.Timeout = dialMongodbTimeout
 
+	err = opts.configureDialInfoIfRequired(dialInfo)
```
Comment (arvenil) [inline shared/connection.go]: I'm not a big fan of modifying values pass by reference. It's trickier to follow. Alternative:
```
err, dialServer := opts.dialServerFunc()
if err != nil {
	glog.Errorf("%s", err)
	return nil
}
dialInfo.DialServer = dialServer
```
It works so 👍

**Pull request merged**